src/main.c

```diff
@@ -358,13 +358,21 @@ void message(char *message){
 
   below2G->common.length = len+sizeof(below2G->common); /* +4 for header */
   memcpy(below2G->text,message,len);
+  
+  
+  int *returnCode = (int*)safeMalloc31(1, "rc");
```

Comment:
FYI, this is how this could have been solved
```diff
+  int returnCode;
   __asm(ASM_PREFIX
         " WTO MF=(E,(%[wtobuf])) \n"
-        :
+        : "=NR:r15"(returnCode)
         :[wtobuf]"NR:r1"(&below2G->common)
         :"r0","r1","r15");
 
   FREE_STRUCT31(
     STRUCT31_NAME(below2G)
   );

```
No need to allocate storage because we don't pass it to the service, we operate on it in AMODE64, so it can be above the bar.

Reply:
The compiler will generate the correct instructions based on `=NR:r15`.